src/main/webapp/app/communication/course-notification/course-notification.service.ts

```
@@ -0,0 +1,416 @@
+        });
+    }
+
+    /**
+     * Gets the icon for a notification type.
+     * Falls back to a default icon if type is not recognized.
+     *
+     * @param type - The notification type
+     * @returns The icon for the notification type
+     */
+    public getIconFromType(type: string | undefined) {
+        if (type === undefined || !CourseNotificationService.NOTIFICATION_TYPE_ICON_MAP.hasOwnProperty(type)) {
```

Biome (1.9.4)

[error] Do not access Object.prototype method 'hasOwnProperty' from target object.

It's recommended using Object.hasOwn() instead of using Object.hasOwnProperty().
See MDN web docs for more details.

(lint/suspicious/noPrototypeBuiltins)